### PR TITLE
Update plugin metadata schema

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

- Adds the `grafanaDependency` property to the plugin metadata schema. 
- Deprecates `grafanaVersion`.
- Relaxes the constraint for the route URL from a `uri` to any string value to account for variable interpolation in url.

### CODE REVIEW

**Comment by marcusolsson** [comment]
The JSON Schema has been an effort to document the _current_ plugin.json properties, and is used to generate the [plugins.json reference docs](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/). 

I'm not aware of any efforts to version the plugin.json schema (although I'd like to eventually). AFAIK this PR doesn't make any _changes_ to the schema, but fills in the gaps and tightens the validation.

That being said, I've yet to see a plugin that makes use of the `dependencies.plugins` property, so I'm guessing it's relatively safe to change it?

**Comment by DanCech** [comment]
Ah ok, I thought this was a proposed change rather than an update to reflect reality.

**Comment by marcusolsson** [comment]
Yes, sorry about that, I should've made it clearer in the description.

File: docs/sources/developers/plugins/metadata.md [hunk 1]
```diff
@@ -47,10 +47,11 @@ Plugin dependencies.
 
 ### Properties
 
-| Property         | Type     | Required | Description                                                                                                      |
-|------------------|----------|----------|------------------------------------------------------------------------------------------------------------------|
-| `grafanaVersion` | string   | **Yes**  | Required Grafana version for this plugin, e.g. `6.x.x 7.x.x` to denote plugin requires Grafana v6.x.x or v7.x.x. |
-| `plugins`        | string[] | No       | An array of required plugins on which this plugin depends.                                                       |
+| Property            | Type     | Required | Description                                                                                                                   |
+|---------------------|----------|----------|-------------------------------------------------------------------------------------------------------------------------------|
+| `grafanaDependency` | string   | **Yes**  | Required Grafana version for this plugin, e.g. `>=7.0.0` to denote plugin requires Grafana 7.0.0 or later                     |
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/metadata.md:52]:
```suggestion
| `grafanaDependency` | string   | **Yes**  | Required Grafana version for this plugin, e.g. `>=7.0.0` to denote plugin requires Grafana 7.0.0 or later.                     |
```

**Review by oddlittlebird — APPROVED** [review]
One super minor edit. Otherwise, good to go.

**Review by DanCech — APPROVED** [review]

**Comment by torkelo** [comment]
what was the point of this?

**Comment by torkelo** [comment]
There is no grafanaDependency property in the plugin model
https://github.com/grafana/grafana/blob/master/pkg/plugins/models.go#L103 

Is this property used by grafana.com plugin api?

**Comment by torkelo** [comment]
Looks like this property was added to plugin.json and handled in grafana.com but not to grafana

**Pull request merged**